File: projetoJava/src/Contrato.java
class Contrato implements Imprimivel {
    private String[] partes;

    public Contrato(String[] partes) {
        this.partes = partes;
    }

    public void imprimir() {
        System.out.print("Contrato entre: ");
        for (int i = 0; i < partes.length; i++) {
            System.out.print(partes[i]);
            if (i < partes.length - 1) {
                System.out.print(", ");
            }
        }
        System.out.println();
    }
}

File: projetoGo/contrato.go
package main

import "fmt"

type Contrato struct {
	partes []string
}

// Método Imprimir para a struct Contrato
func (c Contrato) Imprimir() {
	fmt.Printf("Contrato entre: %s\n", join(c.partes, ", "))
}

// Função join para concatenar partes de string com um delimitador
func join(partes []string, delimitador string) string {
	resultado := ""
	for i, parte := range partes {
		if i > 0 {
			resultado += delimitador
		}
		resultado += parte
	}
	return resultado
}
